lib/ABI/FunctionType.cpp

```diff
@@ -314,6 +253,68 @@ class ConversionHelper {
       }
     }
 
+    auto Arguments = distributeArguments(Function.Arguments,
```

Comment:
Using `auto` so much makes reading the code difficult.

Reply:
Yeah, sorry about that. The main "excuse" I can give is that this code was changing a lot while getting developed - so having `auto`s makes it easier to fix/rename/split/combine types since they are all "just work" as long as the semantics didn't change too much.

I'll see where I can replace `auto` with real types (80 character-lines are pretty tight) right before this is ready to be merged in.

Reply:
(keeping this open as I feel like I might have over-autoed it again, will double-check again right before merging)

Reply:
I went through all the commits again and most of `auto`s that could be trivially replaced in new code are gone: https://github.com/revng/revng/compare/8b1a3b7730dd0ecd0c8225049a94120b4bd710f2..9cfbd7a6984267a114f691626d5a0a919766d3ba

P. S. I also removed the commit introducing unused `UniversalGuard`.